src/java/com/media/FileProcessor.java

```diff
@@ -112,6 +128,75 @@ public static FileProcessor VideoProcessor(File rawDest, File standardDest, File
 		return new FileProcessor(rawDest, standardDest, mobileDest, commands);
 	}
 	
+	/** List of files containing character subsets for languages / language groups in /media/src/resources/orthographies/ */
+	static List<String> orthographies = Arrays.asList("DE", "EN", "ES", "FR", "IT", "PT", "ARABIC", "CYRILLIC", "EASTERN_EUROPE", "NORDIC", "numerals");
+
+	static String subsetCmdBase = "pyftsubset $INPUT --unicodes-file=/home/winterwell/media/src/resources/orthographies/$SUBSET.txt --output-file=$OUTPUT;";
+	static String convertCmdBase = "fontforge -lang=ff -c 'Open(\"$INPUT\"); Generate(\"$OUTPUT\")';";
+
+
+	/**
+	 * Run an uploaded font through Python fontTools pyftsubset and FontForge conversion
+	 * to generate a set of much smaller, language-specific, web-ready font files.
+	 * @param rawFont Original font in TTF, OTF, WOFF, WOFF2
+	 * @param fontDir Directory to output all generated fonts
+	 * @return
+	 */
+	public static FileProcessor FontProcessor(File rawFont, File fontDir) {
+		String inputFontPath = rawFont.getAbsolutePath();
+		String baseName = FileUtils.getBasename(rawFont);
+
+		Map<String,File> dests = new HashMap<String, File>();
+
+		List<String> commands = new ArrayList();
+		commands.add("/bin/bash");
+		commands.add("-c");
```

Comment:
@roscoemcinerney What does this do vs normal Proc, and why is it needed? Thanks.

Reply:
No idea! I know the header says I wrote this, but I have a strong feeling this was Mark, I know he wrote a significant amount of our uploader infrastructure. I'm in favour of refactoring, after this release.